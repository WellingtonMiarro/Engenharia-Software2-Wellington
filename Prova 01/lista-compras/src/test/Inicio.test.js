import React from 'react';
import toJson from 'enzyme-to-json';

//ele torna o snapshot ja em formato de texto
import {createSerializer} from 'enzyme-to-json';

import ROTAS from '../Rotas';
import Listas from '../pages/lista/Listas';

// Utilizando para configuração do ENZYME
import { shallow, configure} from 'enzyme';
import  Adapter from 'enzyme-adapter-react-16';


configure({adapter: new Adapter()});
 expect.addSnapshotSerializer(createSerializer({mode:'deep'}));


describe("Teste de renderização do camponente INICIO",()=>{
    it("Verificar se a página Inicial e carregada adequadamente", ()=>{
        const wrapper = shallow(<Listas/>);//retorna Serializado
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});