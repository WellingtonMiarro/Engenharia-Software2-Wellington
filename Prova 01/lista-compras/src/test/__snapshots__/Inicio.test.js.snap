// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de renderização do camponente INICIO Verificar se a página Inicial e carregada adequadamente 1`] = `
<div>
  <Menu
    logo="logo.png"
    paginaAnterior="/"
    titulo="Lista de Compras"
  />
  <div
    className="conteiner"
  >
    <div>
      <p
        id="mensagemNenhumaLista"
      >
        Clique no botão abaixo para cadastrar uma nova lista!
      </p>
      <div
        id="listagem"
      />
      <div
        id="areaBotao"
      >
        <Link
          to="/criarlista"
        >
          <div
            id="botaoNovaLista"
          >
            <img
              alt="Nova lista"
              src="adicionar.png"
            />
          </div>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
